src/rvi.c

```diff
@@ -1562,11 +1562,9 @@ int rviProcessInput(TRviHandle handle, int *fdArr, int fdLen)
 
         json_decref( root );
     }
-        
-    free( buf );
-
 
 exit:
+    if( buf ) free( buf );
```

Comment:
Good point -- that superfluous check appears at a few other points in the codebase so I'll comb through it as I do my next cleanup.